app/services/placements/import_from_csv.rb

```diff
@@ -4,27 +4,45 @@ module Placements
   class ImportFromCsv
     include ServicePattern
 
+    attr_reader :upload, :unmatched_hesa_ids, :unmatched_urns
+
     def initialize(upload_id:)
       @upload = Upload.find(upload_id)
+      @unmatched_hesa_ids = []
+      @unmatched_urns = []
     end
 
-    # Uploaded CSV file should contain hesa_id and urn headers only
     def call
-      table = CSV.parse(upload.file.download, headers: true)
-      table.each do |row|
-        next unless (school = School.find_by(urn: row["urn"]))
-        next unless (trainee = matching_trainee_in_previous_cycle(row["hesa_id"]))
+      # Uploaded CSV file should contain hesa_id and urn headers only
+      placement_data = CSV.parse(upload.file.download, headers: true, header_converters: :symbol)
 
-        Placement.find_or_create_by(trainee:, school:)
+      placement_data.each do |row|
+        if (trainee = matching_trainee_in_previous_cycle(row[:hesa_id]))
+          urn = row[:urn]
+          if valid_unknown_school_urn?(urn)
```

Comment:
```suggestion
          if is_not_an_applicable_school_urn?(urn)
```

Maybe... since it's does that underneath....

Reply:
Don't want to get too much into semantics, but I think 'not applicable school urn' is not the best description of what these are any more. Also not sure 'magic_urn' is quite right either 😆 , but I don't think it's worth changing the underlying terminology right now.
I thought it was useful to make clear they are valid to use in this context for creating a placement.